terraform/kubernetes/main.tf

```diff
@@ -251,24 +299,44 @@ resource "kubernetes_deployment" "deployment_autoneg_controller_manager" {
 
       spec {
         service_account_name             = kubernetes_service_account.service_account.metadata[0].name
-        automount_service_account_token  = true
```

Comment:
Was this removed because terraform now defaults the value to `true`?

While technically not required, it does seem nice to explicitly assert that the service account token is used by this workload. Terraform has already changed this default once (from `false` which is more secure to `true` which matches the kubernetes default) and kubernetes wants to change the default to `false` since it's more secure but hasn't done so because it would be a breaking change.

Reply:
I removed this to match the kubebuilder autogenerated manifests more closely. That said, I'm happy to consider particular cases.

For posterity, I'll sat that these changes were made by hand, and any exceptions we have here could conceivably get overwritten if/when we'd have some automated tooling to do the conversion.

That said, I'll defer to you - would you like to see an explicit statement here affirming the default of `true` for `automount_service_account_token` ?

Reply:
The default in the terraform kubernetes provider was changed in 2.0.0:
https://github.com/hashicorp/terraform-provider-kubernetes/releases/tag/v2.0.0

Reply:
I've readded this explicit declaration.

Reply:
@rwkarg PTAL and resolve if you're satisfied